### ISSUE TITLE
Sometimes encounters spawn on castles even when castle spawn flag is off.

### ISSUE THREAD

**Issue by WakerPT** [oscarpedrero/BloodyEncounters#3]
Encounters shouldn't spawn on castles as per my config file, however I joined and instantly got a jumpscare from a random encounter.
![imagem](https://github.com/oscarpedrero/BloodyEncounters/assets/7804912/07add000-92b2-463b-8d5a-a01ef4dc0ec2)

Settings file and a quick console log:
```
[Info   :KindredCommands] Player Waker disconnected
[Message:     Unity] User '{Steam 2581457417}' disconnected. approvedUserIndex: 0 Reason: LeftGame k_ESteamNetworkingConnectionState_ClosedByPeer
[Message:     Unity] PlatformSystemBase - EndAuthSession platformId: 76561198014294672
[Message:     Unity] EOS.Session - Entering UnregisterUser!
[Message:     Unity] SteamNetworking - SteamNetConnections.ConnectionUpdate: k_ESteamNetworkingConnectionState_None, 2581457417, 131073
[Message:     Unity] EOS.Session - Entering ModifySessionAsync!
[Message:     Unity] SteamServerTransport - Update 2581457417 - k_ESteamNetworkingConnectionState_None
[Message:     Unity] EOS user unregistered from session successfully!
[Message:     Unity] SteamNetworking - SteamNetConnections.ConnectionUpdate: k_ESteamNetworkingConnectionState_Connecting, 1577046742, 131073
[Message:     Unity] SteamServerTransport - Update 1577046742 - k_ESteamNetworkingConnectionState_Connecting
[Message:     Unity] SteamNetworking - SteamNetConnections.ConnectionUpdate: k_ESteamNetworkingConnectionState_FindingRoute, 1577046742, 131073
[Message:     Unity] SteamServerTransport - Update 1577046742 - k_ESteamNetworkingConnectionState_FindingRoute
[Message:     Unity] SteamNetworking - SteamNetConnections.ConnectionUpdate: k_ESteamNetworkingConnectionState_Connected, 1577046742, 131073
[Message:     Unity] EOS.Session - Entering ModifySessionAsync!
[Message:     Unity] SteamServerTransport - Update 1577046742 - k_ESteamNetworkingConnectionState_Connected
[Message:     Unity] PlatformSystemBase - BeginAuthSession for SteamID: 76561198014294672 Result: k_EBeginAuthSessionResultOK
[Message:     Unity] EOS.Session - Entering RegisterUser!
[Message:     Unity] NetEndPoint '{Steam 1577046742}' reconnect was approved. approvedUserIndex: 0 HasLocalCharacter: True Hail Message Size: 330 Version: 16 PlatformId: 76561198014294672 UserIndex: 2 ShouldCreateCharacter: False IsAdmin: False Length: 330
[Message:     Unity] PlatformSystemBase - OnValidateAuthTicketResponse for SteamID: 76561198014294672, Response: k_EAuthSessionResponseOK
[Message:     Unity] PlatformSystemBase - UserHasLicenseForApp for SteamID: 76561198014294672, Owner: 76561198014294672, Result: k_EUserHasLicenseResultHasLicense, UserContentFlags: Halloween2022
[Message:     Unity] EOS user registered to session successfully!
[Info   :BloodyNotify] [02:28:43] ha conectado Waker
[Message:     Unity] User '{Steam 1577046742}' '76561198014294672', approvedUserIndex: 0, Character: 'Waker' connected as ID '2,1', Entity '631548,1'.
[Info   :KindredCommands] Player Waker connected
[Message:BloodyEncounters] [02:28:44] Attempting to start a new encounter for Waker with Headgear Colector
[Info   :BloodyEncounters] [02:28:44] Next match will be shot at 2024-05-25 02:48:19
```

Settings file:
```
[Main]

## Determines whether the random encounter timer is enabled or not.
# Setting type: Boolean
# Default value: true
Enabled = true

## When enabled, players who are in a castle are excluded from encounters
# Setting type: Boolean
# Default value: true
SkipPlayersInCastle = true

## When enabled, players who are in combat are excluded from the random encounters.
# Setting type: Boolean
# Default value: false
SkipPlayersInCombat = true

## Minimum seconds before a new encounter is initiated. This value is divided by the online users count.
# Setting type: Int32
# Default value: 1200
EncounterTimerMin = 900

## Maximum seconds before a new encounter is initiated. This value is divided by the online users count.
# Setting type: Int32
# Default value: 2400
EncounterTimerMax = 1200

## The lower value for the Player level for encounter.
# Setting type: Int32
# Default value: 10
EncounterMinLevel = 30

## System message template for the encounter.
# Setting type: String
# Default value: You have encountered a <color=#daa520>{0}</color>. You have <color=#daa520>{1}</color> seconds to kill it for a chance of a random reward.
EncounterMessageTemplate = A <color=#daa520>{0}</color> has found you. You have <color=#daa520>{1}</color> seconds to kill it for a chance of a random reward.

## System message template for the reward.
# Setting type: String
# Default value: Congratulations. Your reward: <color={0}>{1}</color>.
RewardMessageTemplate = Congratulations. Your reward: <color={0}>{1}</color>.

## System message template for the reward announcement.
# Setting type: String
# Default value: {0} earned an encounter reward: <color={1}>{2}</color>.
RewardAnnouncementMessageTemplate = {0} earned an encounter reward: <color={1}>{2}</color>.

## If enabled, all online admins are notified about encounters and rewards.
# Setting type: Boolean
# Default value: true
NotifyAdminsAboutEncountersAndRewards = true

## When enabled, all online players are notified about any player's rewards.
# Setting type: Boolean
# Default value: false
NotifyAllPlayersAboutRewards = true
```

**Comment by oscarpedrero** [comment]
@WakerPT The configuration file you are sending me is not from bloodyEncounters, it is from BloodyNotify...

**Comment by WakerPT** [comment]
Oh you're right! 🤦
I've updated my initial report.

**Comment by oscarpedrero** [comment]
Its your castle of its a castle of one member of your clan?!

**Comment by WakerPT** [comment]
My own castle.
I am currently not even part of any clan.

Edit: If it makes any difference, I was playing on the server and we had "solo" mode enabled (max clan size of 1). I had created a clan for myself. I was on it by myself.

A couple of days ago we changed the max clan size to 4 and more recently I deleted my clan.
Not sure if that messed it up somehow? But I think I deleted my clan at least 1 day or so before this happened.
